pkgs/tools/misc/idevicerestore/default.nix

```diff
@@ -0,0 +1,56 @@
+{ stdenv, fetchFromGitHub, autoreconfHook, pkgconfig
+, curl
+, libimobiledevice
+, libirecovery
+, libzip
+, libusbmuxd
+}:
+
+stdenv.mkDerivation rec {
+  pname = "idevicerestore";
+  version = "2019-02-14";
+
+  name = "${pname}-${version}";
+
+  src = fetchFromGitHub {
+    owner = "libimobiledevice";
+    repo = pname;
+    rev = "8a882038b2b1e022fbd19eaf8bea51006a373c06";
+    sha256 = "17lisl7ll43ixl1zqwchn7jljrdyl2p9q99w30i6qaci71mas37m";
+  };
+
+  nativeBuildInputs = [
+    autoreconfHook
+    pkgconfig
+
+    curl
+    libimobiledevice
+    libirecovery
+    libzip
+    libusbmuxd
+    # Inhereting other dependencies from `libimobiledevice`
+  ];
+
+  meta = with stdenv.lib; {
+    homepage = https://github.com/libimobiledevice/idevicerestore;
+    description = "Restore/upgrade firmware of iOS devices";
+    longDescription = ''
+      The idevicerestore tool allows to restore firmware files to iOS devices.
+
+      It is a full reimplementation of all granular steps which are performed during
+      restore of a firmware to a device.
+
+      In general, upgrades and downgrades are possible, however subject to
+      availability of SHSH blobs from Apple for signing the firmare files.
+
+      To restore a device to some firmware, simply run the following:
+      $ sudo idevicerestore -l
+
+      This will download and restore a device to the latest firmware available.
+    '';
+    license = licenses.lgpl3;
```

Comment:
Source code specifies a different license https://github.com/libimobiledevice/idevicerestore/blob/master/src/asr.c#L12

Reply:
@worldofpeace  Hmm, the license file https://github.com/libimobiledevice/idevicerestore/blob/8a882038b2b1e022fbd19eaf8bea51006a373c06/COPYING says LGPL v3.

What do we do when files in the upstream projects have different per-file licenses? Also, given that the one you linked says `or (at your option) any later version`, isn't it still accurate to use `license = licenses.lgpl3`?

Reply:
Addressed by switching to `lgpl21Plus` as suggested further down.